app/labor/markdown_fixer.rb

```diff
@@ -47,7 +68,28 @@ def split_tags(markdown)
     end
 
     def underscores_in_usernames(markdown)
-      markdown.gsub(/(@)(_\w+)(_)/, '\1\\\\\2\\\\\3')
+      return markdown unless markdown.include?("@")
+
+      lines = markdown.dup.lines
+      walker = MarkdownWalker.new(false, lines.first.include?("```"))
+
+      lines.each do |line|
+        walker.walk(line.include?("```"))
+
+        if walker.in_codeblock?
+          line
+        else
+          escape_underscored_username_in_line!(line)
+        end
+      end.join
+    end
+
+    USERNAME_WITH_UNDERSCORE_REGEXP = /(?<!`)@_\w+_/.freeze
```

Comment:
can we add a comment that narrates this regexp? As it's not particularly intuitive